tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -0,0 +1,273 @@
+        thr_iam = threading.Thread(target=self.ha_obj.iam_bucket_cruds,
+                                   args=(event, ), kwargs=args)
+        thr_iam.daemon = True  # Daemonize thread
+        self.test_io_prefix = 'test-45498-io'
+        args = {'s3userinfo': list(self.s3_clean.values())[0], 'log_prefix': self.test_io_prefix,
+                'nclients': 1, 'nsamples': 8, 'output': io_output, 'setup_s3bench': False}
+        LOGGER.info("Step 2.2: Start IOs in background during control pod soft-failure.")
+        io_thread = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        io_thread.daemon = True  # Daemonize thread
+        io_thread.start()
+        thr_iam.start()
+        LOGGER.info("Step 2: Started independent background threads for IOs on already created "
+                    "user and IAM user creation/deletion.")
+
+        LOGGER.info("Step 3: Perform soft-failure of %s control pods in loop and check cluster "
+                    "has failures", self.repl_num - 1)
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        event.set()
+        for pod in pod_list:
+            self.node_master_list[0].kill_process_in_container(pod_name=pod,
+                                                               container_name=const.CORTX_CSM_POD,
+                                                               process_name=const.CSM_AGENT_PRC,
+                                                               safe_kill=True)
+            LOGGER.info("Step 3.1: Check cluster has failures.")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 3.1: Checked cluster has failures.")
+        LOGGER.info("Step 3: Performed soft-failure of %s control pods in loop and checked cluster "
+                    "has failures", self.repl_num - 1)
+
+        time.sleep(self.csm_soft_delay)
+        event.clear()
+        LOGGER.info("Waiting for background threads to join")
+        thr_iam.join()
+        io_thread.join()
+        LOGGER.info("Step 4: Verifying responses from background processes")
+        LOGGER.info("Step 4.1: Verify background process for IAM user creation")
+        iam_resp = tuple()
+        while len(iam_resp) != 4:
+            iam_resp = iam_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not iam_resp:
+            assert_utils.assert_true(False, "Background process failed to do IAM user CRUD "
+                                            "operations")
+        failed = iam_resp[1]
+        exp_failed = iam_resp[0]
+        user_del_failed = iam_resp[2]
+        created_users = iam_resp[3]
+        assert_utils.assert_true(failed, "No IAM user creation/deletion expected to fail before "
```

Comment:
Isn't this check should be assert_false(failed)?
We don't expect anything in failed.